airbyte-integrations/connectors/source-open-exchange-rates/source_open_exchange_rates/source.py

```diff
@@ -0,0 +1,119 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import pendulum
+from pendulum import DateTime
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class OpenExchangeRates(HttpStream, ABC):
+    url_base = "https://openexchangerates.org/api/"
+
+    primary_key = None
+    date_field_name = "timestamp"
+    cursor_field = date_field_name
+
+
+    def __init__(self, base: Optional[str], start_date: str, app_id: str) -> None:
+        super().__init__()
+
+        self.base = base
+        self.start_date = start_date
+        self.app_id = app_id
+        self._cursor_value = None
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+
+        params = {}
+
+        if self.base is not None:
+            params["base"] = self.base
+
+        return params
+
+    def request_headers(self, **kwargs) -> MutableMapping[str, Any]:
+        headers = {"Authorization": f"Token {self.app_id}", "Content-Type": "application/json"}
```

Comment:
You should use the `TokenAuthenticator` instead of defining it yourself.
(It accepts an `auth_method` parameter that you can set to have `Token` instead of `Bearer`)

See [the doc about auth](https://docs.airbyte.com/connector-development/cdk-python/http-streams#authentication)